.github/workflows/build-docker.yaml

```diff
@@ -0,0 +1,33 @@
+name: Build and push spellcheck image
+
+on:
+  push:
+    branches:
+      - main
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: alexslemonade/spellcheck
+
+jobs:
+  build_and_push:
+    runs-on: ubuntu-22.04
+    permissions:
+      contents: read
+      packages: write
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Docker login
+        uses: docker/login-action@v3
+        with:
+          registry: ${{ env.REGISTRY }}
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Build and push to Docker
+        uses: docker/build-push-action@v5
+        with:
+          push: true
+          tags: user/app:latest
```

Comment:
This is a string literal, so not what you want the tag to be. 

I would use `docker/metadata-action` to automatically set this. It's worth it even for small potatoes.

Reply:
I ported over both the tags and labels here after the metadata action, but wasn't entirely sure if we needed them both?